### PR TITLE
Serve openapi.yaml statically.

### PR DESCRIPTION
**Pull request by mikecdavis**
## Summary
- Embeds openapi.yaml into go executable
- Serves openapi.yaml from api router
- Adds basic index handler

### CODE REVIEW

File: pkg/routers/api.go [hunk 1]
```diff
@@ -130,4 +132,12 @@ func WithAPIRouter(opt *APIOptions, r chi.Router) {
...
 
 	r.With(createAccesstokenTimer).Post("/oauth/token", opt.oAuthHandler)
+
+	statikFS, err := fs.New()
+	if err != nil {
+		panic(err)
+	}
+
+	staticServer := http.FileServer(statikFS)
+	r.Handle("/openapi.yaml", staticServer)
```
Comment (tylerbrandt) [inline pkg/routers/api.go]: Is this a conventional place to look for "what is the API spec of this service"? If not, is there such a place (something akin to https://en.wikipedia.org/wiki/List_of_/.well-known/_services_offered_by_webservers but for OpenAPI)? If not, would we be willing to serve it on "/" (index)? Seems a little circular that I have to "just know" this one route in order to discover the rest of the routes I can use.

Comment (mikecdavis) [reply pkg/routers/api.go]: From https://swagger.io/specification/
> It is RECOMMENDED that the root OpenAPI document be named: openapi.json or openapi.yaml.

I also found other examples in their [documentation](https://swagger.io/docs/specification/api-host-and-base-path/#relative-urls) like:
> For example, if the definition hosted at http://localhost:3001/openapi.yaml...

Comment (mikecdavis) [reply pkg/routers/api.go]: That being said I think it's a great idea to have an index "/" handler that could say "hello" and link to the spec. I'll add that as it was also mentioned by @asaschachar as a nice to have to improve the overall "landing page" experience.

File: Makefile [hunk 2]
```diff
@@ -43,22 +43,26 @@ clean: check-go ## runs `go clean` and removes the bin/ dir
...
-cover: check-go ## runs test suite with coverage profiling
+cover: check-go static ## runs test suite with coverage profiling
 	$(GOTEST) ./... -coverprofile=$(COVER_FILE)
 
 cover-html: cover ## generates test coverage html report
 	$(GOCMD) tool cover -html=$(COVER_FILE)
 
 install: check-go ## installs all dev and ci dependencies, but does not install golang
 	curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(GOPATH)/bin v1.19.0
+	go get github.com/rakyll/statik
```
Comment (mjc1283) [inline Makefile]: Why isn't this automatically installed like other deps?

Comment (mikecdavis) [reply Makefile]: This is a tool to embed static files into the binary. This command is installing the `statik` CLI which is run under the `static` make target.

Comment (pawels-optimizely) [reply Makefile]: why do we need this package?  can we server directly using staticServer := http.FileServer(http.Dir(filesDir)) ?

Comment (mikecdavis) [reply Makefile]: So we can still deploy a static binary and not have to incorporate this content into each deployment. Excerpt from the statik repo:
> Is this a crazy idea? No, not necessarily. If you're building a tool that has a Web component, you typically want to serve some images, CSS and JavaScript. You like the comfort of distributing a single binary, so you don't want to mess with deploying them elsewhere. If your static files are not large in size and will be browsed by a few people, statik is a solution you are looking for.

File: web/static/index.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+
+    <title>Optimizely Agent</title>
+    <meta name="description" content="Optimizely Agent">
+    <meta name="author" content="Optimizely">
+
+    <link rel="stylesheet" href="styles.css">
```
Comment (mjc1283) [inline web/static/index.html]: styles.css doesn't exist?

File: web/static/index.html [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+</head>
+
+<body>
+<h1>Welcome to Optimizely Agent!</h1>
```
Comment (asaschachar) [inline web/static/index.html]: Yes! This is exactly what this project needed!

File: web/static/index.html [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    .#######^            '"#@@@@@##|#@@@####"`
+     #####   ,##m           ;|5#T"*@###5\
+     '#M   ;######M     ,;####`  ,#####
+         ;######"   ;######M   ;######`
+        \#####"   s######"   ;######"
+          "#^  ,#######"   s######"
+              %######`  ,#####M"
+                  ``   """`
+</pre>
+Please refer to the <a href="/openapi.yaml">openapi.yaml</a> spec for API details.
```
Comment (asaschachar) [inline web/static/index.html]: Cool, I think this is good to keep as one link. We can evaluate whether we need more links, but I actually think keeping this lean is important for now so we don't duplicate documentation efforts across different surfaces

**Pull request merged**